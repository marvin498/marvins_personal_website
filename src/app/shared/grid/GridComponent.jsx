import React from "react";
import {Grid, List} from 'semantic-ui-react';

const GridComponent = content => (
	<Grid.Row padded="Vertically">
		<Grid.Column>
			<Grid.Row>
				{Object.entries(content).map(itemToBePlacedInRow => (
					<List verticalAlign="middle" relaxed bulleted>
						<List.Item>
							{itemToBePlacedInRow[1]}
						</List.Item>
					</List>
				))}
			</Grid.Row>
		</Grid.Column>
	</Grid.Row>
);

export default GridComponent;