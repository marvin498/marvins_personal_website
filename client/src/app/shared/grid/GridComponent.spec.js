import React from 'react';
import {configure} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure( {adapter: new Adapter()} );
import {shallow, mount, render} from 'enzyme';
import GridComponent from "./GridComponent";

describe("GridComponent", () => {
	it("render", () => {
		const component = shallow(<GridComponent />);
		expect(component).toMatchSnapshot();
	});
});