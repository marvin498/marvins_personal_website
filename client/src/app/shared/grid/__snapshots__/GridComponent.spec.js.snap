// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GridComponent render 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <GridComponent />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <GridColumn>
        <GridRow />
      </GridColumn>,
      "padded": "Vertically",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <GridRow />,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [],
        },
        "ref": null,
        "rendered": Array [],
        "type": [Function],
      },
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <GridColumn>
          <GridRow />
        </GridColumn>,
        "padded": "Vertically",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <GridRow />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [],
          },
          "ref": null,
          "rendered": Array [],
          "type": [Function],
        },
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
